### PR TITLE
*Fix RBAC permission in ABProcessParticipant

### PR DESCRIPTION
**Pull request by wongpratan**
https://github.com/appdevdesigns/planning/issues/341

### CODE REVIEW

File: assets/opstools/AppBuilder/OP/user.js [hunk 1]
```diff
@@ -1,29 +1,59 @@
...
 
       // get the user list
       if (this.userList == null) {
-         Comm.Service.get({ url: "/app_builder/user/list" }).then((data) => {
-            this.userList = data;
-         });
+         tasks.push(
+            new Promise((resolve, reject) => {
+               Comm.Service.get({ url: "/app_builder/user/list" })
+                  .catch(reject)
```
Comment (Hiro-Nakamura) [inline assets/opstools/AppBuilder/OP/user.js:26]: .then().catch()

File: assets/opstools/AppBuilder/OP/user.js [hunk 2]
```diff
@@ -1,29 +1,59 @@
...
       // get the user's scopes
       if (this.scopeList == null) {
-         Comm.Service.get({ url: "/app_builder/user/myscopes" }).then(
-            (data) => {
-               this.scopeList = data;
-            }
+         tasks.push(
+            new Promise((resolve, reject) => {
+               Comm.Service.get({ url: "/app_builder/user/myscopes" })
+                  .catch(reject)
```
Comment (Hiro-Nakamura) [inline assets/opstools/AppBuilder/OP/user.js:40]: .then().catch()

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
just change the order of the ```.catch().then()``` blocks

File: assets/opstools/AppBuilder/OP/user.js [hunk 3]
```diff
@@ -1,29 +1,59 @@
...
+
       // get current user
       if (this.currentUser == null) {
-         Comm.Service.get({ url: "/site/user/data" }).then((data) => {
-            this.currentUser = data.user;
-         });
+         tasks.push(
+            new Promise((resolve, reject) => {
+               Comm.Service.get({ url: "/site/user/data" })
+                  .catch(reject)
```
[the remark this answers is not in the document]
Comment (wongpratan) [reply assets/opstools/AppBuilder/OP/user.js]: Thank you for you review.

File: assets/opstools/AppBuilder/classes/platform/process/ABProcessParticipant.js [hunk 4]
```diff
@@ -15,45 +15,63 @@ function L(key, altText) {
...
+// let __alertCount = 0;
+// let __alertTimeout = 0;
 function loadUsers() {
-   var SiteUser = AD.Model.get("opstools.RBAC.SiteUser");
-   if (SiteUser) {
-      SiteUser.findAll()
-         .fail(function(err) {
-            AD.error.log("ABProcessParticipantCore: Error loading SiteUser", {
+   OP.User.init()
+      .catch(function(err) {
```
[the remark this answers is not in the document]
Comment (wongpratan) [reply assets/opstools/AppBuilder/classes/platform/process/ABProcessParticipant.js:22]: Thank you for you review.

File: assets/opstools/AppBuilder/OP/user.js (hunk 1 above)
Comment (wongpratan) [reply assets/opstools/AppBuilder/OP/user.js]: Thank you for you review.

File: assets/opstools/AppBuilder/OP/user.js (hunk 2 above)
Comment (wongpratan) [reply assets/opstools/AppBuilder/OP/user.js]: Thank you for you review.

**Pull request merged**